### ISSUE TITLE
How to initialize state from json?

### ISSUE THREAD

**Issue by Jebble** [facebookexperimental/Recoil#569]
I've watched and read so much about Recoil already and I get all the simple examples to run, but I can't for the love of me get an initial state to work.

I have a json file with conversations and for every conversation I want to create an atom.
Can somebody please explain how I would go about that?

I'm following the intro's where I have
`const conversationWithId = id => atom({
    key: `conversation-${id}`,
    default: {}
})`

But then how do I loop over my json file and set the atoms to the correct conversation data? Since hooks can't be used inside a callback I simply can't understand how this works.

**Comment by reedom** [comment]
What about to use [atomFamily](https://recoiljs.org/docs/api-reference/utils/atomFamily)?

```js
const yourJsonData = {...};

export const conversationState = atomFamily({
  key: 'conversationState',
  default: id => () => pickConversationFrom(yourJsonData, id)
});

///

function usage() {
  const [threadFoo, setThreadFoo] = useRecoilState(conversationState('foo'));
  const threadBar = useRecoilValue(conversationState('bar'));
}
```

**Comment by Jebble** [comment]
I finally managed, with some help from several YouTube videos, managed to understand the way initializeState and atomFamily works. Really hope the documentation becomes clearer. Now that I know how it works it's great and makes sense, but with just the documentation it was impossible to get here.

**Comment by drarmstr** [comment]
@Jebble - which part of the `initializeState` API was confusing in the docs?

**Comment by Jebble** [comment]
I think it's the combination of Reciol being new and not many examples being given.
I figured it out by reading further into the MutableSnapshot and finding out there that there is the set function you get in intializeState. I just took a lot of tinkering around to realise what was going on and see the way initializeState works. I'm relatively new to the modern JS coming from 10 years of PHP development so I think that also plays a part. 

For me personally a piece of example code showing the importing of an atom and using that snapshot to set the initial value would have made it instantly clear. Once I had it working I was wondering if I'm actually just not so smart..
